src/libraries/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs

```diff
@@ -264,13 +264,20 @@ public override int Read(Span<byte> buffer)
             ThrowIfDisposed();
             if (!CanRead) throw new InvalidOperationException(SR.net_writeonlystream);
 
-            int bytesRead = _streamSocket.Receive(buffer, SocketFlags.None, out SocketError errorCode);
-            if (errorCode != SocketError.Success)
+            try
             {
-                var exception = new SocketException((int)errorCode);
-                throw NetworkErrorHelper.MapSocketException(exception);
+                int bytesRead = _streamSocket.Receive(buffer, SocketFlags.None, out SocketError errorCode);
+                if (errorCode != SocketError.Success)
+                {
+                    var exception = new SocketException((int)errorCode);
+                    throw NetworkErrorHelper.MapSocketException(exception);
```

Comment:
This combined with the catch below is going to result in two NetworkExceptions, one inside the other, isn't it?

Same issue below in ReadByte

Reply:
Yeah, good catch.